golang.md

```diff
@@ -0,0 +1,119 @@
+# Golang at Clever
+
+## Golang versions
+
+All golang services (with some exceptions) at Clever should be on go-1.5.
+
+
+## Docker Images
+
+#### drone
+
+`clever/drone-go:1.5`
+
+#### runtime
+
+  - `gliderlabs/alpine:3.2` - see [catapult/Makefile](https://github.com/Clever/catapult/blob/master/Makefile) as an example. (extra build configuration needed)
+  - `google/debian:wheezy`
+
+
+## Dependencies
+
+Both `godeps` and `glide` are used to manage golang dependencies.
+
+### Glide
+
+Glide is a tool for managing golang dependencies. You can use it to update, pin, and fetch dependencies. Some of the advantages of glide are:
+  - Supports semantic versioning
+  - No need to checkin the vendor directory
+    - Allows per-package repository specification
+    - Drone can use this to fetch private clever repositories!
+  - Warns about dependency/version conflicts
+    - i.e. two different versions of a pkg are being requested
+  - Has fewer bugs than `godep` (or so it seems)
+
+
+#### Initializing a Glide Project
+
+```bash
+> export GO15VENDOREXPERIMENT=1
+> glide init
+> vim `glide.yaml` # set the package name (github.com/Clever/project)
+```
+
+You now have a `glide.yaml` with no dependencies specified. See the next sections on adding dependencies.
+
+#### Converting From Godeps
+
+```bash
+> glide import godep > tmp.yaml
+> mv tmp.yaml glide.yaml
+> rm -rf Godeps/ vendor/
+```
+
+At this point you now have a glide equivalent of your `Godeps/godeps.json` file. Dependencies should be listed and versions pinned where appropriate in the `glide.yaml`. You are now ready to install dependencies. See the next step.
+
+#### Using Glide With in Project
+
+Add `Makefile` rules for downloading glide and installing the project's dependencies:
+```make
+export GO15VENDOREXPERIMENT=1
+
+...
+
+$(GOPATH)/bin/glide:
+    @go get github.com/Masterminds/glide  
+
+install_deps: $(GOPATH)/bin/glide
+    @$(GOPATH)/bin/glide install
+```
+
+##### Download Project Dependencies
+You can now run `make install_deps` to fetch the dependencies. They will be written into `vendor/`. **NOTE** You no longer need to checkin the `vendor` directory!
```

Comment:
We are already using glide in a few repositories and generally it seems nicer. I would personally advocate for using glide over godeps, but thats not my call. I think the go-guild should be making decisions like this. Until we have a guild in place, I think we should at least have a place to start documenting the processes that we are already using so that its easier for others to follow. And thats the intent of this document.